includes/admin/class-edd-notices.php

```diff
@@ -310,6 +313,57 @@ public function show_notices() {
 		settings_errors( 'edd-notices' );
 	}
 
+	/**
+	 * Show a notice if debugging is enabled in the EDD settings.
+	 * Does not show if only the `EDD_DEBUG_MODE` constant is defined.
+	 *
+	 * @since 2.11.5
+	 * @return void
+	 */
+	private function show_debugging_notice() {
+		if ( defined( 'WP_DEBUG' ) && WP_DEBUG ) {
+			return;
+		}
+		if ( ! current_user_can( 'manage_shop_settings' ) ) {
+			return;
+		}
+		if ( ! edd_get_option( 'debug_mode', false ) ) {
+			return;
+		}
+		$view_url = add_query_arg(
+			array(
+				'post_type' => 'download',
+				'page'      => 'edd-tools',
+				'tab'       => 'debug_log',
+			),
+			admin_url( 'edit.php' )
+		);
+		?>
+		<div id="edd-debug-log-notice" class="notice notice-warning">
+			<p>
+				<?php esc_html_e( 'Easy Digital Downloads debug logging is enabled. Please only leave it enabled for as long as it is needed for troubleshooting.', 'easy-digital-downloads' ); ?>
+			</p>
+			<p>
+				<a class="button button-secondary" href="<?php echo esc_url( $view_url ); ?>"><?php esc_html_e( 'View Debug Log', 'easy-digital-downloads' ); ?></a>
+				<button class="button button-primary" id="edd-disable-debug-log"><?php esc_html_e( 'Delete Log File and Disable Logging', 'easy-digital-downloads' ); ?></button>
+			</p>
+		</div>
+		<?php
+	}
+
+	/**
+	 * Disables the debug log setting and deletes the existing log file.
+	 *
+	 * @since 2.11.5
+	 * @return void
+	 */
+	public function edd_disable_debugging() {
+		edd_update_option( 'debug_mode', false );
```

Comment:
Can you add a capability check here for `current_user_can( 'manage_shop_settings' )` ?

Also let's add a nonce.